### PR TITLE
Homepage Posts

### PR DESCRIPTION
**Pull request by vishalg8454**
This PR adds Posts to the Homepage. 
The components added are-
- `Post` at src/components/post
- `Portal` at src/components/portal
- `NewPost` at src/components/new-post
- `SearchBar` at src/components/search-bar
- `Navigation` at src/components/navigation

The pages added are
- `HomePage` at src/pages/home-page

[live preview](https://deploy-preview-2--stalk-me.netlify.app/)

Would love to hear your feedback and suggestions.

image

image

### CODE REVIEW

File: src/components/new-post/NewPost.jsx [hunk 1]
```diff
@@ -1,21 +1,98 @@
...
+        <ReactQuill
+          value={text}
+          style={{ border: "none" }}
+          className="w-full m-2 p-1"
+          fontSize="40px"
+          modules={noModules}
+          onChange={(e) => setText(e)}
+          placeholder={"What's on your mind?"}
+          ref={inputReference}
+        />
```
Comment (TejasShekar) [inline src/components/new-post/NewPost.jsx:61]: Interesting. Did not know about this text editor component. Nice.

**Review by TejasShekar — APPROVED** [review]
Good code implementation. UI also looks clean. Liked the use of quill and other pre-built components.

File: src/components/new-post/NewPost.jsx (hunk 1 above)
Comment (vishalg8454) [reply src/components/new-post/NewPost.jsx:61]: Thanks.

File: src/components/new-post/NewPost.jsx [hunk 2]
```diff
@@ -1,21 +1,98 @@
...
+
+  const noModules = {
+    toolbar: {
+      container: "#toolbar",
+    },
+  };
+
+  useEffect(() => {
+    console.log(text);
+  }, [text]);
```
Comment (Developers-codz) [inline src/components/new-post/NewPost.jsx:31]: remove this code if no longer needed

File: src/components/new-post/NewPost.jsx [hunk 3]
```diff
@@ -1,21 +1,98 @@
...
+          style={{ border: "none" }}
+          id="toolbar"
+          className="border-2 border-blue-500 self-center text-slate-700 flex gap-5 ml-16"
+        >
+          <button className="ql-image">
+            <div>
+              <InsertPhotoOutlinedIcon sx={{ fontSize: 32 }} />
+            </div>
+          </button>
+          <button className="">
```
Comment (Developers-codz) [inline src/components/new-post/NewPost.jsx:75]: can remove the classname

File: src/components/portal/Portal.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const [coordinates, setCoordinates] = useState({});
+
+  const getCoordinates = () => {
+    const rect = anchorRef.current.getBoundingClientRect();
+    const coords = {
+      left: rect.x + rect.width / 2, 
+      top: rect.y, 
+    };
+    return coords;
+  };
```
Comment (Developers-codz) [inline src/components/portal/Portal.jsx:15]: this is something new to me. What portal component is supposed to do ?

File: src/components/portal/Portal.jsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  const updateCoordinates = () => {
+    console.log("updating cords");
```
Comment (Developers-codz) [inline src/components/portal/Portal.jsx:18]: remove console

**Review by Developers-codz — APPROVED** [review]
Code quality is good added few suggestions  you can merge

File: src/components/new-post/NewPost.jsx (hunk 2 above)
Comment (vishalg8454) [reply src/components/new-post/NewPost.jsx:31]: sure.

File: src/components/new-post/NewPost.jsx (hunk 3 above)
Comment (vishalg8454) [reply src/components/new-post/NewPost.jsx:75]: sure.

File: src/components/portal/Portal.jsx (hunk 4 above)
Comment (vishalg8454) [reply src/components/portal/Portal.jsx:15]: Render to a complete different DOM element.

File: src/components/portal/Portal.jsx (hunk 5 above)
Comment (vishalg8454) [reply src/components/portal/Portal.jsx:18]: sure.

**Pull request merged**